### DISCUSSION TITLE
Query elements within an iFrame?

### DISCUSSION THREAD

**Discussion by BEKScheel** [ChromiumDotNet/WebView2.DevTools.Dom#17] [General]
Hi there,

I'm triyng to find elements with QuerySelectorAsync that are located within an iframe on the website. Unfortunately no matter which selectors I try, the elements can't be found (always returns null).  I can, however, find the iframe itself. Is there any solution to this?

Thanks in advance,
blessings,

Kornelius

**Comment by amaitland** [comment]
Is the `iframe` from a different origin?

Did you try running your query directly calling [document.querySelector)](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) in the `DevTools console` to confirm it is indeed correct?

Scripts are by default executed against the `MainFrame`. you should be able to access any of the child frames via `DevToolsContext.Frames` property.

See https://github.com/ChromiumDotNet/WebView2.DevTools.Dom/blob/main/WebView2.DevTools.Dom.Tests/FrameTests/FrameManagementTests.cs

**Reply by BEKScheel** [reply]
Thank You so much! That solved it! :-) 
Accessing it via DevToolsContext.Frames works perfectly! :-)

**Reply by thibchap** [reply]
How about when the iframe is from a different origin? Is it even possible?
I've tried calling `QuerySelectorAsync` on the corresponding `DevToolsContext.Frames`, but the method never returns!

With the `CoreWebView2.FrameCreated` event, I can get reference on the `CoreWebView2Frame` instance and call the `ExecuteScriptAsync` method for that frame. But with the `WebView2.DevTools.Dom` library, it only works on same origin frames.
